### DISCUSSION TITLE
Deprecating macros

### DISCUSSION THREAD

**Discussion by pstaabp** [openwebwork/pg#731] [General]
There has been discussion about deprecating some macros from PG either because the functionality
1. doesn't work anymore OR
2. the functionality is obsolete OR
3.  the functionality is not core to PG.

Here is a list of macros to deprecate:
- unionProblem.pl (no functionality)
- unionInclude.pl (problem randomizing is a problem and this requires `$probNum` to be defined in the problem)
- LiveGraphicsCylindricalPlot3D.pl (no longer working)
- LiveGraphicsParametricCurve3D (no longer working)
- LiveGraphicsParametricSurface3D (no longer working)
- LiveGraphicsRectangularPlot3D (no longer working)
- LiveGraphicsVectorField2D (no longer working)
- LiveGraphicsVectorField3D (no longer working)

**Comment by pstaabp** [comment]
A few others:
- compoundProblem.pl
- compoundProblem2.pl
- compoundProblem5.pl

These are all superseded by scaffold.pl

**Comment by drgrice1** [comment]
Yeah, those certainly should be considered deprecated.

Also, in your original post you included the LiveGraphics macros, and state that they are no longer working.  Those are all working again.
